_config.yml

```diff
@@ -9,6 +9,7 @@
 title: Edit _config.yml to set your title!
 description: This is where the description of your site will go. You should change it by editing the _config.yml file. It can be as long as you like! Happy blogging... ❤
 github_username: jph00
+github_repo: blog # used for puslishing links back to your repo in the blog.
```

Comment:
does the addition of this break anything?

Reply:
As a new setting, the only thing that would be broken as you pointed out is the github and colab links, unless users actually go and update their `_config.yml`. 

If they use a different repo name, and then use `{% github_notebook_link %}`, the page would build, but the link would be broken. 

I think once someone starts investigating, they'll find the setting in a few minutes (since the snippet users `{{site.github_repo}}`right after `{{site.github_user}}`, and it's a common Jekyll config lookup pattern. But ideally, this can be documented, and thus it'll be even easier for users. 

One alternative is to do `github_repo : %CHANGE_ME_IN_CONFIG_YML%` — that would be broken as well, but that would point people to where they should fix it. I think `about.md` does this, and `description:` setting by default says something in the nature of `This is where the description text is used, change me in config.yml`. 

Should I change it to something descriptive like that?

Reply:
Oh I see.   In this case you might not need the the `github_repo` variable at all.  This template comes pre-loaded with the [github-metadata](https://github.com/jekyll/github-metadata) plugin, which allows you to reference the full path as

`{{ site.url }}{{ site.baseurl }}`

For example, [I am doing this for word docs](https://github.com/fastai/fast_template/blob/master/_action_files/word2post.sh#L27)